src/quicklist.c

```diff
@@ -1051,8 +1169,11 @@ int quicklistDelRange(quicklist *quicklist, const long start,
 }
 
 /* Passthrough to ziplistCompare() */
-int quicklistCompare(unsigned char *p1, unsigned char *p2, int p2_len) {
-    return ziplistCompare(p1, p2, p2_len);
+int quicklistCompare(quicklistEntry* entry, unsigned char *p2, const size_t p2_len) {
+    if(unlikely(entry->node->container == QUICKLIST_NODE_CONTAINER_NONE)) {
+        return  (strncmp((char *)entry->value ,(char *)p2,sdslen((char*)p2)) == 0);
```

Comment:
1) do you mean that len1 = entry->sz in case of plain node ? 
2) there is a comment here - /* Passthrough to ziplistCompare() */ should I remove it completely or change it ?

Reply:
the comment is no longer relevant, in the past it meant that there's no reason to describe what this function does, since it's just a wrapper. but that's no longer true, so a proper comment is probably needed.

regarding the memcmp, yes, there's no point in doing memcmp if the buffers are not the same size (i.e. we know the buffers are not equal). and obviously ignoring the size of one of them and doing memcmp with the size of the other, could cause a segmentation fault.